modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -1650,6 +1648,9 @@ void GDScriptAnalyzer::resolve_function_signature(GDScriptParser::FunctionNode *
 			return_type.type_source = GDScriptParser::DataType::INFERRED;
 			return_type.kind = GDScriptParser::DataType::VARIANT;
 			p_function->set_datatype(return_type);
+#ifdef DEBUG_ENABLED
+			parser->push_warning(p_function, GDScriptWarning::NOT_STATICALLY_TYPED, "Function", p_function->identifier ? p_function->identifier->name : "<anonymous lambda>");
```

Comment:
The `identifier != nullptr` does not guarantee that `identifier->name` is not empty and that function is lambda. There is the correct code higher. Take it out from the `for` loop (wrapped in `#ifdef`), and use it for both `UNUSED_PARAMETER` and `NOT_STATICALLY_TYPED` warnings.

https://github.com/godotengine/godot/blob/fcddb37a45275320ed6c317c0ddc48ddbab2c0f1/modules/gdscript/gdscript_analyzer.cpp#L1597-L1600

```suggestion
			parser->push_warning(p_function, GDScriptWarning::NOT_STATICALLY_TYPED, "Function", function_visible_name);
```